### PR TITLE
Update main readme to include data generation.

### PR DESCRIPTION
**Pull request by lior-dikstein**
Updated data generation output loss function to last linear nodes. Upated extractor to find last linear node.

### CODE REVIEW

File: model_compression_toolkit/data_generation/pytorch/model_info_exctractors.py [hunk 1]
```diff
@@ -153,24 +154,26 @@ def _set_hooks_for_last_layers(self):
...
-                        self.output_layers_hooks.update({name: hook})
-                        hook_handle = module.register_forward_hook(hook.hook)  # Register the hook on the module
-                        self.hook_handles.append(hook_handle)
+
+            # Find the output nodes in the graph
+            if node.op == 'output':
+                found_linear_node = False
+                nodes_to_search = node.all_input_nodes
+
+                # Search graph from the output node and back untill we find a linear node
```
Comment (elad-c) [inline model_compression_toolkit/data_generation/pytorch/model_info_exctractors.py:164]: until

File: model_compression_toolkit/data_generation/pytorch/optimization_functions/output_loss_functions.py [hunk 2]
```diff
@@ -67,18 +67,21 @@ def regularized_min_max_diff(
...
     output_loss = torch.zeros(1).to(get_working_device())
 
     for output_weight, output, last_layer_input in zip(weights_output_layers, output_imgs, output_layers_inputs):
-        weights_norm = torch.linalg.norm(output_weight, dim=1)
-        output = torch.reshape(output, [output.shape[0], -1])
+        weights_norm = torch.linalg.norm(output_weight.squeeze(), dim=1)
         out_max, out_argmax = torch.max(output, dim=1)
         out_min, out_argmin = torch.min(output, dim=1)
-        last_layer_norm = torch.linalg.norm(last_layer_input, dim=1)
+        last_layer_avg = torch.reshape(last_layer_input, [last_layer_input.shape[0], last_layer_input.shape[1], -1])
```
Comment (elad-c) [inline model_compression_toolkit/data_generation/pytorch/optimization_functions/output_loss_functions.py:82]: less characters: last_layer_input.shape[:2] + [-1]

**Review by elad-c — APPROVED** [review]

File: model_compression_toolkit/data_generation/README.md [hunk 3]
```diff
@@ -18,7 +18,10 @@ pip install model-compression-toolkit
 
 ## Usage
 ### PyTorch
+Explore a Jupyter Notebook example showcasing data generation with ResNet18, including visualizations, and a practical example of Post Training Quantization:
+* [Data Generation for Resnet18 with PyTorch](tutorials/notebooks/example_pytorch_data_generation.ipynb).
```
Comment (ofirgo) [inline model_compression_toolkit/data_generation/README.md:22]: verify link because I cannot access it (getting page not found)

**Review by ofirgo — APPROVED** [review]
